### DISCUSSION TITLE
Unify files across a hierarchy from Go

### DISCUSSION THREAD

**Discussion by danielpcox** [cue-lang/cue#1586] [Q&A] [answered]
- I'm trying to use some CUE from Go, organized in a directory hierarchy for clarity.
- I have a directory hierarchy like in this screenshot, which shows subdirectories of `cuemodule/`. Also, `cuemodule/cue.mod/module.cue` contains `module: greymatter.io/operator`. 
![image](https://user-images.githubusercontent.com/56411/158477646-499f9aab-af18-4aab-b943-827fa872a667.png)
- I'd like to unify things all across the hierarchy. For concreteness, say I'm trying to unify these:
    - `inputs.cue`
    - `intermediates.cue`
    - `k8s/intermediates.cue`
    - `k8s/outputs/control.cue`
- Every one of those files has `package only` in there.
- This CLI command works to get what I want: `cue eval ./only/k8s/outputs -p only` and it seems to walk up the directory hierarchy to collect everything necessary, but I'm having trouble replicating this behavior from Go.
- In Go, I've tried using `cuelang.org/go/cue/load`'s `load.Instances` with different `load.Config` options, but the errors I get suggest that it either takes issue with the package not matching the directory name (saying, e.g., that it wanted `output` but found `only`, or that it couldn't find things up or down the hierarchy depending on how I tweak things.
- I suspect I could make this work by changing the `package` declarations at the top of each file to match the parent directory names, and then refer to all attributes with their package namespace, but I'd prefer not to do that if at all possible, and the CLI made me believe it was possible. However, if I do have to do this, is there something equivalent to Python's `from whatever import *` so I could avoid peppering package refs all over the place?

Am I missing something obvious?

**Comment by verdverm** [comment]
Nothing stands out as obvious. Your best bet will be to post a working example back here, feel free to edit your original discussion text.

https://github.com/cue-lang/cue/wiki/Creating-test-or-performance-reproducers

**Comment by danielpcox** [comment, marked as answer]
I was able to work around this issue by specifying a relative path (rooted in `.`) for the `args`, to `load.Instances`, and a `load.Config` with a `Dir` that was an absolute path to the CUE module directory. I'm still not sure why specifying the package declaration (with a module prefix) didn't work as `args`, but I'm not bothered about it.
